lib/Input.js

```diff
@@ -37,7 +47,9 @@ class Input extends React.Component {
   }
 
   componentDidUpdate (prevProps) {
-    if (this.props.autoresize && prevProps.query !== this.props.query) {
+    if (prevProps.autoresize !== this.props.autoresize ||
+      (this.props.autoresize && RESIZE_PROPS.some((prop) => this.props[prop] !== prevProps[prop]))
+    ) {
```

Comment:
That could lead to an infinite loop, though, because `setState` will get called every time `componentDidUpdate` is called, which will cause a re-render, which in turn causes `componentDidUpdate` to be called again, where we'd call `setState` and cause a re-render, etc...  

So taking that approach requires a `shouldComponentUpdate` implementation which should see if any of the underlying state or props _actually_ requires a re-render (or extend from `React.PureComponent` instead of `React.Component`, but then you need React 15.3, and the current dependency is 15.2).

So long story short, that will actually make this patch a bit more complicated than it currently is.